flake.nix

```diff
@@ -0,0 +1,221 @@
+{
+  description = "BDK Flake to run all tests locally and in CI";
+
+  inputs = {
+    # stable nixpkgs (let's not YOLO on unstable)
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.11";
+
+    # pin dependencies to a specific version
+    # find instructions here:
+    # <https://lazamar.co.uk/nix-versions>
+
+    # bitcoind pinned to 0.25.1
+    nixpkgs-bitcoind.url = "github:nixos/nixpkgs?rev=53793ca0aecf67164c630ca34dbfde552a8ab085";
+    # TODO: pin to 0.26.0 once it stops to fail in darwin
+    # pinned to 0.26.0
+    # nixpkgs-bitcoind.url = "github:nixos/nixpkgs?rev=f5375ec98618347da6b036a5e06381ab7380db03";
+
+    # Blockstream's esplora
+    # inspired by fedimint CI
+    nixpkgs-kitman.url = "github:jkitman/nixpkgs?rev=61ccef8bc0a010a21ccdeb10a92220a47d8149ac";
+
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs = {
+        nixpkgs.follows = "nixpkgs";
+        flake-utils.follows = "flake-utils";
+      };
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+
+  outputs = { self, nixpkgs, nixpkgs-bitcoind, nixpkgs-kitman, rust-overlay, flake-utils, pre-commit-hooks, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        overlays = [ (import rust-overlay) ];
+        lib = pkgs.lib;
+        stdenv = pkgs.stdenv;
+        isDarwin = stdenv.isDarwin;
+        libsDarwin = with pkgs.darwin.apple_sdk.frameworks; lib.optionals isDarwin [
+          # Additional darwin specific inputs can be set here
+          Security
+          SystemConfiguration
+          CoreServices
+        ];
+
+        # Dependencies
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+        pkgs-bitcoind = import nixpkgs-bitcoind {
+          inherit system overlays;
+        };
+        pkgs-kitman = import nixpkgs-kitman {
+          inherit system;
+        };
+
+        # Signed Commits
+        signed-commits = pkgs.writeShellApplication {
+          name = "signed-commits";
+          runtimeInputs = [ pkgs.git ];
+          text = builtins.readFile ./ci/commits_verify_signature.sh;
+        };
+
+        # Toolchains
+        # latest stable
+        stable = pkgs.rust-bin.stable.latest.default.override {
+          targets = [ "wasm32-unknown-unknown" ]; # wasm
+        };
```

Comment:
The Rust version 1.63.0 is the minimum supported in msrv. It is good to set the default shell with this version, else developer will easily write code that can not pass the clippy.